### PR TITLE
Allow policies to affect all contacts.

### PR DESCRIPTION
**Pull request by tobiasraabe**
### Current behavior

Contact policies apply only to the result of a single contact model. 

### Desired behavior

The `"affected_contact_model"` entry of a contact policy dict becomes optional. If not present, the policy is called with the full contacts DataFrame and also has to return a modified copy of the full contacts DataFrame. This allows to have correlated behavior (i.e. careful individuals reduce all contacts when they have symptoms) without adding another type variable to the state DataFrame. 

Importantly, the changes are backwards compatible!

### Before / After Comparison:

image

### CODE REVIEW

**Review by janosg — APPROVED** [review]

File: src/sid/contacts.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def calculate_contacts(
 
         if model["is_recurrent"]:
             model_specific_contacts = model_specific_contacts.astype(bool)
+        else:
```
Comment (tobiasraabe) [inline src/sid/contacts.py:58]: I am not sure if I like it. Of course, we can prevent some unnecessary errors, but at the same time it conflicts with a user's expectation if she created an integer. Even worse, memory consumption is increased. Maybe we could make it optional if the dtype is `object` or fix sid-germany altogether

File: src/sid/contacts.py [hunk 2]
```diff
@@ -618,7 +620,8 @@ def post_process_contacts(contacts, states, contact_models):
 
     # Dead people and ICU patients don't have contacts.
     has_no_contacts = states["needs_icu"] | states["dead"]
-    contacts.loc[has_no_contacts] = 0
+    contacts.loc[has_no_contacts, random_models] = 0
+    contacts.loc[has_no_contacts, recurrent_models] = False
```
Comment (tobiasraabe) [inline src/sid/contacts.py:624]: Sorry, I thought the zero is automatically converted to False.

Comment (roecla) [reply src/sid/contacts.py:624]: Yes, that is right. We did this, trying to solve the typing Error we had that turned out to be caused elsewhere and left it in. It shouldn't make a difference.

File: src/sid/contacts.py (hunk 1 above)
Comment (roecla) [reply src/sid/contacts.py:58]: I will address this in a new PR.

**Pull request merged**